### PR TITLE
En_Gb2

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_En_Gb2/z_en_gb2.c [hunk 1]
```diff
@@ -46,98 +50,983 @@ const ActorInit En_Gb2_InitVars = {
...
+                    this->unk_26E = 0x14D6;
+                    this->unk_26C |= 2;
+                    func_801518B0(globalCtx, this->unk_26E, &this->actor);
+                    break;
+            }
+        } else if (this->unk_26E == 0x14DA) {
+            switch (globalCtx->msgCtx.choiceIndex) {
+                case 0:
+                    func_8019F208();
+                    func_801159EC(this->unk_288 * -1);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Gb2/z_en_gb2.c:435]: Out of curiosity, would this match?
```suggestion
                    func_801159EC(-this->unk_288);
```

File: src/overlays/actors/ovl_En_Gb2/z_en_gb2.c [hunk 2]
```diff
@@ -46,98 +50,983 @@ const ActorInit En_Gb2_InitVars = {
...
+        switch (globalCtx->msgCtx.choiceIndex) {
+            case 0:
+                if (gSaveContext.rupees < this->unk_288) {
+                    play_sound(NA_SE_SY_ERROR);
+                    this->unk_26E = 0x14D7;
+                    this->unk_26C |= 2;
+                    func_801518B0(globalCtx, this->unk_26E, &this->actor);
+                } else {
+                    func_8019F208();
+                    func_801159EC(this->unk_288 * -1);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Gb2/z_en_gb2.c:587]: Here too
```suggestion
                    func_801159EC(-this->unk_288);
```

**Review by AngheloAlf — COMMENTED** [review]
Just some minor comments

File: src/overlays/actors/ovl_En_Gb2/z_en_gb2.c (hunk 1 above)
Comment (Kelebek1) [reply src/overlays/actors/ovl_En_Gb2/z_en_gb2.c]: I thought this would be change codegen so I didn't actually try, but yeah it works no problem.

**Review by AngheloAlf — APPROVED** [review]

File: src/overlays/actors/ovl_En_Gb2/z_en_gb2.c [hunk 3]
```diff
@@ -46,98 +50,983 @@ const ActorInit En_Gb2_InitVars = {
...
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Gb2/func_80B10B5C.s")
+s32 EnGb2_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx,
+                           Gfx** gfx) {
+    EnGb2* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Gb2/func_80B10DAC.s")
+    if (limbIndex == 7) {
+        limbIndex = limbIndex;
+        Matrix_RotateY(this->unk_270.y, 1);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Gb2/z_en_gb2.c:987]:
```suggestion
        Matrix_RotateY(this->unk_270.y, MTXMODE_APPLY);
```

**Review by engineer124 — COMMENTED** [review]
Good work. Just two minor comments

File: src/overlays/actors/ovl_En_Gb2/z_en_gb2.c [hunk 4]
```diff
@@ -46,98 +48,984 @@ const ActorInit En_Gb2_InitVars = {
...
+}
+
+void func_80B11268(EnGb2* this, GlobalContext* globalCtx) {
+    if (globalCtx->roomCtx.currRoom.num == 1) {
+        this->unk_290 = 0;
+        this->unk_282[0] = this->actor.cutscene;
+        if (Actor_GetRoomCleared(globalCtx, 2) && Actor_GetRoomCleared(globalCtx, 3) &&
+            Actor_GetRoomCleared(globalCtx, 4) && Actor_GetRoomCleared(globalCtx, 5)) {
+            this->unk_28A = 0xFF;
+            this->unk_26C &= 0xFEFF;
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Gb2/z_en_gb2.c:846]:
```suggestion
            this->unk_26C &= ~0x100;
```

**Review by hensldm — APPROVED** [review]
Small cosmetic suggestion

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**